go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus/prometheus
 
-go 1.22.7
+go 1.23.0
```

Comment:
We have a policy to maintain the same versions as upstream Go. Which means we can not yet update this to Go 1.23 since Go 1.22 is still supported until Go 1.24 is released.

Unfortunately, upstream [Kubernetes KEP-3744](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/3744-stay-on-supported-go-versions) has a policy that conflicts with this.

So we can not update this for now.